argus/client/base.py

```diff
@@ -0,0 +1,170 @@
+        return {
+            "schema_version": self.schema_version
+        }
+
+    @property
+    def request_headers(self):
+        return {
+            "Authorization": f"token {self.auth_token}",
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }
+
+    def get(self, endpoint: str, location_params: dict[str, str] = None, params: dict = None) -> requests.Response:
+        response = requests.get(
+            url=self.get_url_for_endpoint(
+                endpoint=endpoint,
+                location_params=location_params
+            ),
+            params=params,
+            headers=self.request_headers
+        )
+
+        return response
+
+    def post(
+        self,
+        endpoint: str,
+        location_params: dict = None,
+        params: dict = None,
+        body: dict = None,
+    ) -> requests.Response:
+        response = requests.post(
+            url=self.get_url_for_endpoint(
+                endpoint,
+                location_params=location_params
+            ),
+            params=params,
+            json=body,
+            headers=self.request_headers,
+        )
+
+        return response
+
+    def submit_run(self, run_type: str, run_body: dict) -> requests.Response:
+        return self.post(endpoint=self.Routes.SUBMIT, location_params={"type": run_type}, body={
+            **self.generic_body,
+            **run_body
+        })
+
+    def set_status(self, run_type: str, run_id: UUID, new_status: TestStatus) -> requests.Response:
+        return self.post(
+            endpoint=self.Routes.SET_STATUS,
+            location_params={"type": run_type, "id": str(run_id)},
+            body={
+                **self.generic_body,
+                "new_status": new_status.value
+            }
+        )
+
+    def update_product_version(self, run_type: str, run_id: UUID, product_version: str) -> requests.Response:
+        return self.post(
+            endpoint=self.Routes.SET_PRODUCT_VERSION,
+            location_params={"type": run_type, "id": str(run_id)},
+            body={
+                **self.generic_body,
+                "product_version": product_version
+            }
+        )
+
+    def submit_logs(self, run_type: str, run_id: UUID, logs: list[LogLink]) -> requests.Response:
+        return self.post(
+            endpoint=self.Routes.SUBMIT_LOGS,
+            location_params={"type": run_type, "id": str(run_id)},
+            body={
+                **self.generic_body,
+                "logs": [asdict(l) for l in logs]
+            }
+        )
+
+    def finalize_run(self, run_type: str, run_id: UUID) -> requests.Response:
+        return self.post(
+            endpoint=self.Routes.FINALIZE,
+            location_params={"type": run_type, "id": str(run_id)},
+            body={
+                **self.generic_body,
+            }
+        )
+
+    def heartbeat(self, run_type: str, run_id: UUID) -> requests.Response:
```

Comment:
still returning response while it is not necessary.

Reply:
Oops, missed this one

Reply:
Nevermind, that's a base client method, returning a response here is fine as it's consumed by `sct_heartbeat` method.

Reply:
why not to move `self.check_response(response)` to base client method and don't return anything?